File: android/app/src/main/java/com/choicecrafter/students/models/ModuleProgress.java
package com.choicecrafter.studentapp.models;

public class ModuleProgress {
    private int completedTasks;
    private int totalTasks;

    public ModuleProgress() {
        // Default constructor required for Firestore serialization
    }

    public ModuleProgress(int completedTasks, int totalTasks) {
        this.completedTasks = completedTasks;
        this.totalTasks = totalTasks;
    }

    public int getCompletedTasks() {
        return completedTasks;
    }

    public void setCompletedTasks(int completedTasks) {
        this.completedTasks = completedTasks;
    }

    public int getTotalTasks() {
        return totalTasks;
    }

    public void setTotalTasks(int totalTasks) {
        this.totalTasks = totalTasks;
    }

    public float getCompletionPercentage() {
        return totalTasks == 0 ? 0 : (completedTasks * 100f) / totalTasks;
    }

    public boolean isCompleted() {
        return totalTasks > 0 && completedTasks >= totalTasks;
    }
}

File: lib/models/module_progress.dart
class ModuleProgress {
  ModuleProgress({
    required this.completedTasks,
    required this.totalTasks,
  });

  final int completedTasks;
  final int totalTasks;

  double get completionPercentage {
    if (totalTasks == 0) {
      return 0;
    }
    return (completedTasks * 100) / totalTasks;
  }

  bool get isCompleted => totalTasks > 0 && completedTasks >= totalTasks;
}
